### ISSUE TITLE
`clojure-lsp dump`: Allow user to provide clj-kondo analysis configuration

### ISSUE THREAD

**Issue by sheluchin** [clojure-lsp/clojure-lsp#1384]
**Is your feature request related to a problem? Please describe.**
In some cases LSP users may want additional clj-kondo analysis data which LSP currently does not ask for nor need. 

**Describe the solution you'd like**
Allow a [clj-kondo analysis config](https://github.com/clj-kondo/clj-kondo/tree/master/analysis#extra-analysis) map to be passed in when running the `clojure-lsp dump` command. This map should be merged with clojure-lsp's [existing analysis config](https://github.com/clojure-lsp/clojure-lsp/blob/master/lib/src/clojure_lsp/kondo.clj#L236-L255). 

Adding the option for such user configuration should not conflict with LSP's current operations, because it just extends the data that clj-kondo returns. 

**Additional context**
There is a bit of discussion about this idea in this Slack [thread](https://clojurians.slack.com/archives/CPABC1H61/p1668960512798619).

**Comment by ericdallo** [comment]
This should not be hard to implement, any suggestion for the flag name to dump command?

**Comment by sheluchin** [comment]
@ericdallo if brevity is not much of a concern `--clj-kondo-analysis-config` seems fairly self-explanatory. Otherwise maybe something like `--analysis-config`, `--analysis-overrides`, `--clj-kondo-config-extras`, `--clj-kondo-analysis` and so on.

**Comment by ericdallo** [comment]
I´d like to use the existent `--analysis EDN` option that now exists after previous commit, WDYT? so maybe we could have `--analysis '{:project-type :project-and-dependencies :clj-kondo-overrides {...}}'`

**Comment by sheluchin** [comment]
Yeah, good point, @ericdallo. That previous ticket already provides the `--analysis` key to give the clj-kondo context to options, so extending the map here is a good idea 👍
